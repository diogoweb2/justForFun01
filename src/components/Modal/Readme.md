```js
<Modal>test Modal</Modal>
```
