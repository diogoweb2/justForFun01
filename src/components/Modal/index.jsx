import React from "react";
import PropTypes from "prop-types";
import Button from "../Button";

const Modal = ({ title, children, handleVisibility, show }) => {
  if (!show) {
    return (
      <Button onClick={handleVisibility} type="link">
        {title}
      </Button>
    );
  }
  return (
    <div>
      <section
        role="dialog"
        tabIndex="-1"
        aria-labelledby="modal-heading-01"
        aria-modal="true"
        aria-describedby="modal-content-id-1"
        className="slds-modal slds-fade-in-open"
      >
        <div className="slds-modal__container">
          <header className="slds-modal__header">
            <h2 className="slds-text-heading_medium slds-hyphenate">{title}</h2>
          </header>
          <div className="slds-modal__content slds-p-around_medium">
            {children}
          </div>
          <footer className="slds-modal__footer">
            <Button onClick={handleVisibility} type="secundary">
              Close
            </Button>
          </footer>
        </div>
      </section>
      <div className="slds-backdrop slds-backdrop_open" />
    </div>
  );
};

Modal.propTypes = {
  /**
   * Modal Title
   */
  title: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  /**
   * event to handle open/close
   */
  handleVisibility: PropTypes.func.isRequired,
  /**
   * show or hide
   */
  show: PropTypes.bool
};

Modal.defaultProps = {
  show: false
};

export default Modal;
