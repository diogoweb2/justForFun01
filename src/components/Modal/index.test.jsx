import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Modal from "./index";

const fn = jest.fn();
let wrapper;

describe("Modal", () => {
  it("should render", () => {
    wrapper = shallow(<Modal>test</Modal>);
    expect(wrapper).toHaveLength(1);
  });

  it("should match snapshot opened...", () => {
    wrapper = shallow(
      <Modal title="test title" show>
        test
      </Modal>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should match snapshot closed...", () => {
    wrapper = shallow(
      <Modal title="test title" show={false}>
        test
      </Modal>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
