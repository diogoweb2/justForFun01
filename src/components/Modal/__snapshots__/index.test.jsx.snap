// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should match snapshot closed... 1`] = `
<Button
  type="link"
>
  test title
</Button>
`;

exports[`Modal should match snapshot opened... 1`] = `
<div>
  <section
    aria-describedby="modal-content-id-1"
    aria-labelledby="modal-heading-01"
    aria-modal="true"
    className="slds-modal slds-fade-in-open"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="slds-modal__container"
    >
      <header
        className="slds-modal__header"
      >
        <h2
          className="slds-text-heading_medium slds-hyphenate"
        >
          test title
        </h2>
      </header>
      <div
        className="slds-modal__content slds-p-around_medium"
      >
        test
      </div>
      <footer
        className="slds-modal__footer"
      >
        <Button
          type="secundary"
        >
          Close
        </Button>
      </footer>
    </div>
  </section>
  <div
    className="slds-backdrop slds-backdrop_open"
  />
</div>
`;
